### ISSUE TITLE
SWEP:Deployed(), TOOL:Deployed(), SWEP:Holstered() and TOOL:Holstered() hooks

### ISSUE THREAD

**Issue by WilliamVenner** [Facepunch/garrysmod-requests#1739]
### Details

https://discordapp.com/channels/565105920414318602/589120351238225940/770048866501918771

Prediction can be an annoying issue on the client to deal with when working with SWEPs and TOOLs.

In some cases `SWEP/TOOL:Deploy/Holster()` doesn't get called at all, which can break stuff, or requires a lot of crap to work around it.

I propose the new hooks:

```
SWEP:Deployed()
TOOL:Deployed()
SWEP:Holstered()
TOOL:Holstered()
```

These will be called via receiving a net message or similar means on the client (for example; there might be a [better method](https://discordapp.com/channels/565105920414318602/589120351238225940/770053434517880853)), and simply after their respective unpredicted serverside functions on the server.

Basically, they're guaranteed to be called in both realms when the event happens, even if some delay is necessary (such delay should be documented appropriately.)
